connect/api/src/main/java/org/apache/kafka/connect/transforms/Transformation.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package org.apache.kafka.connect.transforms;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+
+import java.util.Map;
+
+/**
+ * Single message transformation for Kafka Connect record types.
+ *
+ * Connectors can be configured with transformations to make lightweight message-at-a-time modifications.
+ */
+public interface Transformation<R extends ConnectRecord<R>> {
+
+    /** Initialize with the provided configuration properties. **/
+    void init(Map<String, Object> props);
```

Comment:
`Transformation` get initialized with `originalsWithPrefix` which returns a `Map<String, Object>`... in any case I'm thinking of removing this and extending `Configurable` as @hachikuji suggested. `Configurable` specifies:

```
public void configure(Map<String, ?> configs);
```

which works for the `originalsWithPrefix` result value.